clarifaipyspark/dataset.py

```diff
@@ -0,0 +1,361 @@
+    return input_obj.list_inputs(
+        dataset_id=self.dataset_id, input_type=input_type, per_page=per_page)
+
+  def list_annotations(self, per_page: int = None, input_type: str = None):
+    """Lists all the annotations for the inputs in the dataset of a clarifai app.
+
+    Args:
+        per_page (str): No of response of inputs per page.
+        input_type (str): Input type that needs to be displayed (text,image)
+        TODO: Do we need input_type ?, since in our case it is image, so probably we can go with default value of "image".
+
+    Examples:
+        TODO
+
+    Returns:
+        list of annotations.
+    """
+    ### input_ids: list of input_ids for which user wants annotations
+    input_obj = Inputs(user_id=self.user_id, app_id=self.app_id)
+    all_inputs = list(
+        input_obj.list_inputs(
+            dataset_id=self.dataset_id, input_type=input_type, per_page=per_page))
+    return input_obj.list_annotations(batch_input=all_inputs)
+
+  def export_annotations_to_dataframe(self):
+    """Export all the annotations from clarifai App to spark dataframe.
+
+    Examples:
+        TODO
+
+    Returns:
+        spark dataframe with annotations"""
+
+    annotation_list = []
+    spark = SparkSession.builder.appName('Clarifai-spark').getOrCreate()
+    input_obj = Inputs(user_id=self.user_id, app_id=self.app_id)
+    all_inputs = list(input_obj.list_inputs(dataset_id=self.dataset_id))
+    response = list(input_obj.list_annotations(batch_input=all_inputs))
+    for an in response:
+      temp = {}
+      temp['annotation'] = json.loads(MessageToJson(an.data))
+      if not temp['annotation'] or temp['annotation'] == '{}' or temp['annotation'] == {}:
+        continue
+      temp['id'] = an.id
+      temp['user_id'] = an.user_id
+      temp['input_id'] = an.input_id
+      try:
+        created_at = float(f"{an.created_at.seconds}.{an.created_at.nanos}")
+        temp['created_at'] = time.strftime('%m/%d/% %H:%M:%5', time.gmtime(created_at))
+        modified_at = float(f"{an.modified_at.seconds}.{an.modified_at.nanos}")
+        temp['modified_at'] = time.strftime('%m/%d/% %H:%M:%5', time.gmtime(modified_at))
+      except:
+        temp['created_at'] = float(f"{an.created_at.seconds}.{an.created_at.nanos}")
+        temp['modified_at'] = float(f"{an.modified_at.seconds}.{an.modified_at.nanos}")
+      annotation_list.append(temp)
+    return spark.createDataFrame(annotation_list)
+
+  def export_images_to_volume(self, path, input_response):
+    for resp in input_response:
+      imgid = resp.id
+      ext = resp.data.image.image_info.format
+      url = resp.data.image.url
+      img_name = path + '/' + imgid + '.' + ext.lower()
+      headers = {"Authorization": self.metadata[0][1]}
+      response = requests.get(url, headers=headers)
+      with open(img_name, "wb") as f:
+        f.write(response.content)
+
+  def export_text_to_volume(self, path, input_response):
```

Comment:
Can we change to `input_proto` instead. I  assume that's what we are passing `resources_pb2.Input()`

Reply:
it is the response from list_inputs() from dataset class

Reply:
Yes and it is `resources_pb2.Input()` not the endpoint response
https://github.com/Clarifai/clarifai-python/blob/f036545ac4a8a695451003322152e8f8f3219e09/clarifai/client/input.py#L751